atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/CommitLockTest.java

```diff
@@ -0,0 +1,216 @@
+            LockResponse response = acquireCellLock("row1", "col2");
+            assertTrue(response.wasSuccessful());
+        };
+
+        Transaction transaction = startTransaction(canAcquireLockOnDifferentCell, conflictHandler);
+        put(transaction, "row1", "col1", "100");
+        transaction.commit();
+    }
+
+    private Transaction startTransaction(PreCommitCondition preCommitCondition, ConflictHandler conflictHandler) {
+        ImmutableMap<TableReference, ConflictHandler> tablesToWriteWrite = ImmutableMap.of(
+                TEST_TABLE,
+                conflictHandler,
+                TransactionConstants.TRANSACTION_TABLE,
+                ConflictHandler.IGNORE_ALL);
+        return new SerializableTransaction(
+                MetricsManagers.createForTests(),
+                keyValueService,
+                new LegacyTimelockService(timestampService, lockService, lockClient),
+                transactionService,
+                NoOpCleaner.INSTANCE,
+                Suppliers.ofInstance(timestampService.getFreshTimestamp()),
+                TestConflictDetectionManagers.createWithStaticConflictDetection(tablesToWriteWrite),
+                SweepStrategyManagers.createDefault(keyValueService),
+                0L,
+                Optional.empty(),
+                preCommitCondition,
+                AtlasDbConstraintCheckingMode.NO_CONSTRAINT_CHECKING,
+                null,
+                TransactionReadSentinelBehavior.THROW_EXCEPTION,
+                true,
+                timestampCache,
+                AtlasDbConstants.DEFAULT_TRANSACTION_LOCK_ACQUIRE_TIMEOUT_MS,
+                AbstractTransactionTest.GET_RANGES_EXECUTOR,
+                AbstractTransactionTest.DEFAULT_GET_RANGES_CONCURRENCY,
+                MultiTableSweepQueueWriter.NO_OP,
+                MoreExecutors.newDirectExecutorService(),
+                CommitProfileProcessor.createNonLogging(metricsManager)) {
+            @Override
+            protected Map<Cell, byte[]> transformGetsForTesting(Map<Cell, byte[]> map) {
+                return Maps.transformValues(map, input -> input.clone());
+            }
+        };
+    }
+
+    private LockResponse acquireRowLock(String rowName) {
+        LockDescriptor rowLockDescriptor = AtlasRowLockDescriptor.of(
+                TEST_TABLE.getQualifiedName(),
+                PtBytes.toBytes(rowName));
+        return lock(rowLockDescriptor);
+    }
+
+    private LockResponse acquireCellLock(String rowName, String colName) {
+        LockDescriptor cellLockDescriptor = AtlasCellLockDescriptor.of(
+                TEST_TABLE.getQualifiedName(),
+                PtBytes.toBytes(rowName),
+                PtBytes.toBytes(colName));
+        return lock(cellLockDescriptor);
+    }
+
+    private LockResponse lock(LockDescriptor lockDescriptor) {
+        TimelockService timelockService = new LegacyTimelockService(timestampService, lockService, lockClient);
```

Comment:
This works because the underlying lock service is consistent, but we really should just store the single timelock service somewhere